rm/rm-server/src/test/java/functionaltests/nodesource/TestLocalInfrastructureCronPolicy.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package functionaltests.nodesource;
+
+import static functionaltests.utils.RMTHelper.log;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.ow2.proactive.resourcemanager.common.event.RMEventType;
+import org.ow2.proactive.resourcemanager.frontend.ResourceManager;
+import org.ow2.proactive.resourcemanager.nodesource.infrastructure.LocalInfrastructure;
+import org.ow2.proactive.resourcemanager.nodesource.policy.CronPolicy;
+
+import functionaltests.nodesource.helper.CronPolicyTestHelper;
+import functionaltests.nodesource.helper.LocalInfrastructureTestHelper;
+import functionaltests.utils.RMFunctionalTest;
+import functionaltests.utils.RMTHelper;
+
+
+/**
+ * Test of the {@link CronPolicy}. Acquires the nodes of a {@link LocalInfrastructure}
+ * immediately, then wait between one and two minutes before removing them,
+ * and then wait one more minute to reacquire the nodes.
+ */
+public class TestLocalInfrastructureCronPolicy extends RMFunctionalTest {
+
+    private static final int NODES_NUMBER = 2;
+
+    private static final String EMPTY_NODE_SOURCE_NAME = "emptyNodeSource";
+
+    private static final String NODE_SOURCE_NAME = "LocalInfrastructureCronPolicyTestNodeSource";
+
+    private ResourceManager resourceManager;
+
+    @Before
+    public void setup() throws Exception {
+        this.resourceManager = this.rmHelper.getResourceManager();
+    }
+
+    @Test
+    public void action() throws Exception {
+
+        log("Create an remove an empty local node source with cron policy");
+        createEmptyNodeSource(EMPTY_NODE_SOURCE_NAME);
+        removeNodeSource(EMPTY_NODE_SOURCE_NAME);
+
+        log("Create a local node source with " + NODES_NUMBER + " nodes with cron policy");
+        createNodeSource(NODE_SOURCE_NAME);
+
+        log("Waiting for the cron policy to remove the nodes");
+        for (int i = 0; i < NODES_NUMBER; i++) {
```

Comment:
maybe you use Lambde.repeater

```
repeater.accept(NODES_NUMBER, () -> { 
  his.rmHelper.waitForAnyNodeEvent(RMEventType.NODE_REMOVED);
}
```

Reply:
Thanks. I applied it in most of infrastructure and policy tests.